treeherder/model/management/commands/cycle_data.py

```diff
@@ -82,40 +81,27 @@ class PerfherderCycler(DataCycler):
     def __init__(self, days, chunk_size, sleep_time, is_debug=None,
                  logger=None, **kwargs):
         super().__init__(days, chunk_size, sleep_time, is_debug, logger)
-        self.unsheriffed_repos_expire_days = datetime.timedelta(days=kwargs['days_for_nonmain'])
         self.started_at = None
 
-    @property
-    def should_quit(self):
-        now = datetime.datetime.now()
-        elapsed_runtime = now - self.started_at
-
-        return self.max_runtime < elapsed_runtime
-
     def cycle(self):
         self.started_at = datetime.datetime.now()
+        max_chunk_size = self.chunk_size
 
         try:
-            for repository in Repository.objects.all():
-                if self.should_quit:
-                    break
-                self.logger.warning('Cycling repository: {0}'.format(repository.name))
-                PerformanceDatum.objects.cycle_data(repository,
-                                                    self.cycle_interval,
-                                                    self.chunk_size,
-                                                    self.sleep_time,
-                                                    self.unsheriffed_repos_expire_days,
-                                                    self.logger,
-                                                    self.started_at,
-                                                    self.max_runtime)
+            PerformanceDatum.objects.cycle_data(self.cycle_interval,
+                                                max_chunk_size,
+                                                self.logger,
+                                                self.started_at,
+                                                self.max_runtime)
         finally:
-            # TODO-igoldan: use info if visible on production
-            elapsed_seconds = (datetime.datetime.now() - self.started_at).total_seconds()
-            hours = elapsed_seconds // 3600
-            seconds = (elapsed_seconds // 60) % 60
+            self.report_runtime()
 
-            self.logger.warning('Performance data cycling lasted for {}h{}s'
-                                .format(hours, seconds))
+    def report_runtime(self):
```

Comment:
Is this still useful? I think we'd be able to see how long it's been running from the Papertrail logs and based on what the start time is set in the Heroku scheduler.

Reply:
Yes, that's a good point.